module/zcommon/zfs_fletcher.c

```diff
@@ -380,18 +380,52 @@ void
 fletcher_4_incremental_native(const void *buf, uint64_t size,
     zio_cksum_t *zcp)
 {
+	const fletcher_4_ops_t *ops;
+	uint64_t p2size = P2ALIGN(size, 64);
+
 	ASSERT(IS_P2ALIGNED(size, sizeof (uint32_t)));
 
-	fletcher_4_scalar_native(buf, size, zcp);
+#ifndef _KERNEL
```

Comment:
You should be able to check `fletcher_4_initialized` in user and kernel space without the `#ifndef _KERNEL`.  You will then get the optimized versions when running with ztest.